maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/invoker/equipped/InvokerEquippedEmbeddedMavenRunningAsDaemonTestCase.java

```diff
@@ -0,0 +1,55 @@
+package org.jboss.shrinkwrap.resolver.impl.maven.embedded.invoker.equipped;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.concurrent.TimeoutException;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.Invoker;
+import org.assertj.core.api.Assertions;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.Utils.getPropertiesWithSkipTests;
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.Utils.pathToJarSamplePom;
+
+public class InvokerEquippedEmbeddedMavenRunningAsDaemonTestCase {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
+
+    @Test
+    public void testWhenDaemonIsUsedEndOfTheBuildIsNotReached() throws InterruptedException, TimeoutException {
+
+        final InvocationRequest request = new DefaultInvocationRequest();
+        Invoker invoker = new DefaultInvoker();
+
+        request.setPomFile(new File(pathToJarSamplePom));
+        request.setGoals(Arrays.asList(new String[] { "clean", "verify" }));
+
+        request.setProperties(getPropertiesWithSkipTests());
+
+        EmbeddedMaven
+            .withMavenInvokerSet(request, invoker)
+            .useAsDaemon()
+            .build();
+
+        Thread.sleep(1000);
```

Comment:
The end of the build should not have been reached - the 1 sec is only for printing the started string. And this string is printed by Resolver as part of the build invocation, so I believe that it should be enough.